### PR TITLE
Improvement: Use max resolution and correct aspect ratio for iPhone's playlist toggle button

### PR DESCRIPTION
**Pull request by wutschel**
## Description

The cover thumb used for iPhone's playlist toggle button was not using the maximum (native) pixel resolution of the display and a wrong aspect ratio.

Screenshots (white background for debugging):
<a href="https://ibb.co/QnCHwZG">Bildschirmfoto-2024-10-12-um-21-49-01</a>

## Summary for release notes

Improvement: Use max resolution and correct aspect ratio for iPhone's playlist toggle button

### CODE REVIEW

File: XBMC Remote/NowPlaying.m [hunk 1]
```diff
@@ -297,27 +297,29 @@ - (void)setPlaylistCellProgressBar:(UITableViewCell*)cell hidden:(BOOL)hidden {
...
-	CGContextDrawImage(bitmap, CGRectMake(destWidth / 2 - width / 2, destHeight / 2 - height / 2, width, height), imageRef);
-	CGImageRef ref = CGBitmapContextCreateImage(bitmap);
-    UIImage *result = [UIImage imageWithCGImage:ref scale:image.scale orientation:image.imageOrientation];
-	
-	CGContextRelease(bitmap);
-    CGColorSpaceRelease(colorSpace);
-	CGImageRelease(ref);
-	
-	return result;
+    @autoreleasepool {
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:300]: that's a useless change here - no objects to autorelease

File: XBMC Remote/NowPlaying.m [hunk 2]
```diff
@@ -307,10 +307,15 @@ - (UIImage*)resizeImage:(UIImage*)image width:(int)destWidth height:(int)destHei
...
+        CGContextRef bitmap = CGBitmapContextCreate(NULL,
+                                                    destWidth,
+                                                    destHeight,
+                                                    8,
+                                                    4 * destWidth,
+                                                    colorSpace,
+                                                    kCGBitmapAlphaInfoMask & kCGImageAlphaPremultipliedLast);
         
-        CGContextDrawImage(bitmap, CGRectMake(destWidth / 2 - width / 2, destHeight / 2 - height / 2, width, height), imageRef);
+        CGContextDrawImage(bitmap, CGRectMake(floor(destWidth - width) / 2, floor(destHeight - height) / 2, width, height), imageRef);
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:318]: why not flooring the final result?

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: XBMC Remote/NowPlaying.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m]: Removed, but did a re-indent instead.

File: XBMC Remote/NowPlaying.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m]: Thanks for pointing me to this, this was intended but coded wrong.

**Pull request merged**